modyn/supervisor/internal/triggers/datadrifttrigger.py

```diff
@@ -127,39 +137,67 @@ def inform(
             data stream where the model's performance may have started to degrade due to drift."""
 
         new_key_ts = [(key, timestamp) for key, timestamp, _ in new_data]
-        detect_interval = self.config.detection_interval_data_points
-        offset = self._total_items_in_current_window % detect_interval
-
-        if offset + len(new_key_ts) < detect_interval:
-            # No detection in this trigger
-            self._update_curr_window(new_key_ts)
-            return
-
-        # At least one detection, fill up window up to that detection
-        self._update_curr_window(new_key_ts[: detect_interval - offset])
-        new_key_ts = new_key_ts[detect_interval - offset :]
-        trigger_idx = detect_interval - offset - 1  # If we trigger, it will be on this index
-
-        if not self._triggered_once:
-            # If we've never triggered before, always trigger
-            self._triggered_once = True
-            triggered = True
-            drift_results: dict[str, MetricResult] = {}
-        else:
-            # Run the detection
-            triggered, drift_results = self._run_detection()
-
-        yield from self._handle_drift_result(triggered, trigger_idx, drift_results, log=log)
+        initial_batch_pointer = 0
 
         # Go through remaining data in new data in batches of `detect_interval`
-        for i in range(0, len(new_key_ts), detect_interval):
-            batch = new_key_ts[i : i + detect_interval]
-            trigger_idx += detect_interval
-            self._update_curr_window(batch)
-
-            if len(batch) == detect_interval:
-                # Regular batch, in this case run detection
-                triggered, drift_results = self._run_detection()
+        while True:
+            if self._sample_left_until_detection - len(new_key_ts) > 0:
+                # No detection in this trigger because of too few data points to fill detection interval
+                self._windows.inform_data(new_key_ts)  # update current window
+                self._sample_left_until_detection -= len(new_key_ts)
+                return
+
+            # At least one detection, fill up window up to that detection
+            next_detection_interval = new_key_ts[: self._sample_left_until_detection]
+            self._windows.inform_data(next_detection_interval)
+
+            # Update the remaining data
+            initial_batch_pointer += len(next_detection_interval)
+            new_key_ts = new_key_ts[self._sample_left_until_detection :]
+
+            # The first ever detection will always trigger
+            if not self._triggered_once or len(self.warmup_intervals) < (self.config.warmup_intervals or 0):
+                # If we've never triggered before, always trigger
+                self._triggered_once = True
+                triggered = True
+                drift_results: dict[str, MetricResult] = {}
+                self.warmup_intervals.append(list(self._windows.reference_))
```

Comment:
why do we always append this, even if not using warmups?

Reply:
good catch, simply a bug